tests/reference/tests/chocolateysource/apply_manifest_without_location.rb

```diff
@@ -0,0 +1,24 @@
+require 'chocolatey_helper'
+test_name 'MODULES-3037 - Add Source: Apply manifest without location'
+confine(:to, :platform => 'windows')
+
+backup_config
+
+# arrange
+chocolatey_src = <<-PP
+  chocolateysource {'chocolatey':
+    ensure   => present,
+  }
+PP
+
+# teardown
+teardown do
+  reset_config
+end
+
+# act
+step 'Apply Manifest'
+apply_manifest(chocolatey_src, :catch_failures => true) do
+  step 'Verify Result'
+  assert_match(/Notice: \/Stage\[main\]\/Main\/Chocolateysource\[chocolatey\]\/location: location changed 'https:\/\/chocolatey.org\/api\/v2\/' to 'chocolatey'/, stdout, "stdout did not match expected")
```

Comment:
Yeah, I understand that bit.  The path as title makes sense (and we saw that looking through the code).

However, if you already have a choco source defined that is *not* a UNC path, HTTP uri or local path on disk... then I don't think it makes sense to do something like this:


```
chocolateysource { 'foo': }
```

If a source named `foo` exists, it sets location to `foo`, which is probably *not* what you wanted - it obliterates the location with something unusable.

The question is... would it ever make sense for a user to do this:

```
chocolateysource { 'c:\\packages':
  location => 'https://foo.com/'
}
```

That's allowable as far as I can tell, but arguably doesn't make much sense, right?